core/src/main/java/org/jdbi/v3/core/transaction/LocalTransactionHandler.java

```diff
@@ -135,7 +135,7 @@ public boolean isInTransaction(Handle handle) {
             if (!didTxnRollback.get()) {
                 handle.commit();
             }
-        } catch (Exception e) {
+        } catch (Throwable e) {
```

Comment:
From the Throwable javadoc:

```
For the purposes of compile-time checking of exceptions, {@code
Throwable} and any subclass of {@code Throwable} that is not also a
subclass of either {@link RuntimeException} or {@link Error} are
regarded as checked exceptions.
```

So I don't understand how the compiler allows `throw e` in this context.

Reply:
My money is on generics erasure shenanigans confusing the compiler. Report it as a bug and win 5 million bucks for finding a flaw in java?

Reply:
Explicitly writing ` throw new Throwable();` anywhere in the method still gives a compile error...

Reply:
Exhibit A ^^

![image](https://user-images.githubusercontent.com/3950300/51709031-a83d9e00-2025-11e9-9ec8-08e501570f85.png)

Reply:
Yeah, looks like the compiler can narrow the caught exception type on its own regardless of the declaration:
![image](https://user-images.githubusercontent.com/3950300/51709376-7d077e80-2026-11e9-9175-8c34deb05e07.png)

So it probably omits the "can't throw a Throwable here" because it knows you'll never catch a `Throwable`.

Reply:
Huh. Well that's nifty.

Reply:
For completeness:
![image](https://user-images.githubusercontent.com/3950300/51710228-be009280-2028-11e9-9f3c-00193a31cf0d.png)